### PR TITLE
Update Flux dependencies

### LINKED ISSUE #3810: Update staging environment with Flux 2.0.0-rc.5
We need to update the staging environment with Flux 2.0.0-rc.5 and test Weave GitOps to ensure everything works as expected.

The only outstanding issue is https://github.com/weaveworks/weave-gitops/issues/3626 which we should be able replicate and retest in staging once fixed.

### PR DESCRIPTION
**Pull request by yiannistri**
Closes #3810

**What changed?**
Updated Flux dependencies.

**Why was this change made?**
To keep up with Flux changes.

**How was this change implemented?**
Updated the go modules for the Flux v1 APIs. Also updated code to accommodate controller-runtime version bump.

**How did you validate the change?**
Run locally

**Release notes**
N/A

**Documentation Changes**
N/A

### CODE REVIEW

File: core/clustersmngr/cluster/delegating_cache.go [hunk 1]
```diff
@@ -56,17 +61,18 @@ func (c *delegatingCacheCluster) makeCachingClient(leafClient client.Client) (cl
...
-		// https://github.com/kubernetes-sigs/controller-runtime/issues/612
-		// TODO: Research if we can change the way we query those events so we can enable the cache for it.
-		UncachedObjects:   []client.Object{&v1.Event{}},
-		CacheUnstructured: true,
+	cl, err := client.New(c.restConfig, client.Options{
+		Cache: &client.CacheOptions{
+			Reader: delegatingCache,
+			// Non-exact field matches are not supported by the cache.
+			// https://github.com/kubernetes-sigs/controller-runtime/issues/612
+			// TODO: Research if we can change the way we query those events so we can enable the cache for it.
```
Comment (yitsushi) [inline core/clustersmngr/cluster/delegating_cache.go:69]: File an issue about it please, otherwise this comment will live there "forever".

**Review by chanwit — APPROVED** [review]
The codes around `gvks` look good to me.

**Comment by yitsushi** [comment]
I don't think we can fix the `apiVersion` under the `YAML` tab on the resource details page. :disappointed:

Summary:
go-client does not care about GroupVersion only the Kind. On Get and List, the requested object's gvk will be set to be the same as the request was.

If we request `v1beta2` we will get a `v1beta2`, if we request a `v1` we will get a `v1` with the same object. If we iterate over all known group versions, we will get a response on all of them with the same objects. If we request only `v1` (as it does not matter) it will show `v1` even if the object is `v1beta2`.

My assumption on the reason:
You can't have multiple resources with the same Kind, Namespace, and Name, does not matter the `apiVersion`.

---

I went down to a deep rabbit hole with controller-runtime and stopped when I was in the source code of client-go about handling metadata lists. My assumption (and it matches with our experiences) is, the list query does not care about group versions at all.

For example [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/client.go#L378-L388) sets the object's gvk to the requested gvk on list.
It does the same for get.

**Comment by chanwit** [comment]
In that case, we would
1. look for the latest version
2. use it to set a GVK
3. document that Weave GitOps will display object with the latest version of schema in the YAML tab.

wdyt?

**Comment by yitsushi** [comment]
> document that Weave GitOps will display object with the latest version of schema in the YAML tab.

my only issue with that, it can be misleading, not because "oh that line is different", but given version may not have a specific field, or it has a different type. The UI is presented mostly to people without `kubectl` access (I think) because they don't have for security reasons, or the cluster is behind a firewall and only the UI is exposed on a private network address. Yes there is a command above the yaml output, but if someone checks the resources and they try to understand something with the docs open in another tab, they can be confused.

File: cmd/gitops/beta/run/cmd.go [hunk 2]
```diff
@@ -1045,6 +1045,7 @@ func runCommandInnerProcess(cmd *cobra.Command, args []string) error {
 							podErr error
 						)
 
+						//nolint:staticcheck
```
Comment (yitsushi) [inline cmd/gitops/beta/run/cmd.go:1048]: Can you please add a comment why did you add this?
- https://github.com/weaveworks-liquidmetal/flintlock/blob/a19f96e2496757f09076e948c308bc2393bd245f/test/e2e/utils/runner.go#L208
- https://github.com/weaveworks-liquidmetal/flintlock/blob/a19f96e2496757f09076e948c308bc2393bd245f/infrastructure/grpc/server.go#L77

Comment (yitsushi) [reply cmd/gitops/beta/run/cmd.go:1048]:
```suggestion
						//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: core/clustersmngr/factory.go [hunk 3]
```diff
@@ -275,6 +275,7 @@ func (cf *clustersManager) watchClusters(ctx context.Context) {
 
 	cf.initialClustersLoad <- true
 
+	//nolint:staticcheck
```
Comment (yitsushi) [inline core/clustersmngr/factory.go:278]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: core/clustersmngr/factory.go [hunk 4]
```diff
@@ -312,6 +313,7 @@ func (cf *clustersManager) watchNamespaces(ctx context.Context) {
 	// waits the first load of cluster to start watching namespaces
 	<-cf.initialClustersLoad
 
+	//nolint:staticcheck
```
Comment (yitsushi) [inline core/clustersmngr/factory.go:316]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: core/fluxsync/fluxsync.go [hunk 5]
```diff
@@ -44,11 +44,13 @@ func RequestReconciliation(ctx context.Context, k client.Client, name client.Obj
 
 // WaitForSync polls the k8s API until the resources is sync'd, and times out eventually.
 func WaitForSync(ctx context.Context, c client.Client, key client.ObjectKey, obj Reconcilable) error {
+	//nolint:staticcheck
```
Comment (yitsushi) [inline core/fluxsync/fluxsync.go:47]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: core/fluxsync/fluxsync.go [hunk 6]
```diff
@@ -44,11 +44,13 @@ func RequestReconciliation(ctx context.Context, k client.Client, name client.Obj
...
 	if err := wait.PollImmediate(
 		k8sPollInterval,
 		k8sTimeout,
 		checkResourceSync(ctx, c, key, obj, obj.GetLastHandledReconcileRequest()),
 	); err != nil {
+		//nolint:staticcheck
```
Comment (yitsushi) [inline core/fluxsync/fluxsync.go:53]:
```suggestion
		//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/install/install_dashboard.go [hunk 7]
```diff
@@ -206,6 +206,7 @@ func ReconcileDashboard(ctx context.Context, kubeClient client.Client, dashboard
 
 	var sourceRequestedAt string
 
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/install/install_dashboard.go:209]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/install/install_dashboard.go [hunk 8]
```diff
@@ -217,6 +218,7 @@ func ReconcileDashboard(ctx context.Context, kubeClient client.Client, dashboard
 	}
 
 	// wait for the reconciliation of dashboard to be done
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/install/install_dashboard.go:221]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/install/install_dashboard.go [hunk 9]
```diff
@@ -232,6 +234,7 @@ func ReconcileDashboard(ctx context.Context, kubeClient client.Client, dashboard
 	}
 
 	// wait for dashboard to be ready
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/install/install_dashboard.go:237]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/localkubernetes/configure.go [hunk 10]
```diff
@@ -188,6 +189,7 @@ func directConnection(vRawConfig *clientcmdapi.Config, service *corev1.Service,
 
 	server := fmt.Sprintf("https://127.0.0.1:%v", service.Spec.Ports[0].NodePort)
 	var err error
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/localkubernetes/configure.go:192]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/localkubernetes/configure.go [hunk 11]
```diff
@@ -231,6 +233,7 @@ func createProxyContainer(vClusterName, vClusterNamespace string, rawConfig, vRa
 	}
 
 	server := fmt.Sprintf("https://127.0.0.1:%v", localPort)
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/localkubernetes/configure.go:236]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/localkubernetes/configure.go [hunk 12]
```diff
@@ -300,6 +303,7 @@ func CreateBackgroundProxyContainer(vClusterName, vClusterNamespace string, rawC
 		return "", errors.Errorf("error starting background proxy : %s %v", string(out), err)
 	}
 	server := fmt.Sprintf("https://127.0.0.1:%v", localPort)
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/localkubernetes/configure.go:306]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/localkubernetes/configure.go [hunk 13]
```diff
@@ -366,6 +370,7 @@ func getServerFromExistingProxyContainer(vClusterName, vClusterNamespace string,
 		localPort, err := strconv.Atoi(strings.TrimSpace(string(out)))
 		if err == nil && localPort != 0 {
 			server := fmt.Sprintf("https://127.0.0.1:%v", localPort)
+			//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/localkubernetes/configure.go:373]:
```suggestion
			//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/portforward/start.go [hunk 14]
```diff
@@ -37,6 +37,7 @@ func StartPortForwardingWithRestart(config *rest.Config, address, pod, namespace
 			log.Actionf("Restarting port forwarding")
 
 			// wait for loft pod to start
+			//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/portforward/start.go:40]:
```suggestion
			//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/remove.go [hunk 15]
```diff
@@ -42,6 +42,7 @@ func Remove(kubeClient client.Client, session *InternalSession) error {
 		result = multierror.Append(result, err)
 	}
 
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/session/remove.go:45]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/session/remove.go [hunk 16]
```diff
@@ -61,6 +62,7 @@ func Remove(kubeClient client.Client, session *InternalSession) error {
...
```
Comment (yitsushi) [inline pkg/run/session/remove.go:65]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/watch/setup_dev_ks.go [hunk 17]
```diff
@@ -428,6 +428,7 @@ func ReconcileDevBucketSourceAndKS(ctx context.Context, log logger.Logger, kubeC
 	}
 
 	// wait for the reconciliation of dev-bucket to be done
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/watch/setup_dev_ks.go:431]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/watch/setup_dev_ks.go [hunk 18]
```diff
@@ -443,6 +444,7 @@ func ReconcileDevBucketSourceAndKS(ctx context.Context, log logger.Logger, kubeC
 	}
 
 	// wait for devBucket to be ready
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/watch/setup_dev_ks.go:447]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/watch/setup_dev_ks.go [hunk 19]
```diff
@@ -470,6 +472,7 @@ func ReconcileDevBucketSourceAndKS(ctx context.Context, log logger.Logger, kubeC
 		return err
 	}
 
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/watch/setup_dev_ks.go:475]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/run/watch/setup_dev_ks.go [hunk 20]
```diff
@@ -485,6 +488,7 @@ func ReconcileDevBucketSourceAndKS(ctx context.Context, log logger.Logger, kubeC
 	}
 
 	devKs := &kustomizev1.Kustomization{}
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/run/watch/setup_dev_ks.go:491]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: pkg/services/crd/fetcher.go [hunk 21]
```diff
@@ -43,6 +43,7 @@ type defaultFetcher struct {
 }
 
 func (s *defaultFetcher) watchCRDs() {
+	//nolint:staticcheck
```
Comment (yitsushi) [inline pkg/services/crd/fetcher.go:46]:
```suggestion
	//nolint:staticcheck // deprecated, tracking issue: https://github.com/weaveworks/weave-gitops/issues/3812
```

File: core/clustersmngr/cluster/delegating_cache.go (hunk 1 above)
Comment (yiannistri) [reply core/clustersmngr/cluster/delegating_cache.go:69]: https://github.com/weaveworks/weave-gitops/issues/3814

**Review by yitsushi — APPROVED** [review]

**Review by chanwit — APPROVED** [review]
